origin-growth/src/apollo/db.js

```diff
@@ -0,0 +1,41 @@
+const db = require('../models')
+
+function _makeCampaign(row) {
+  let status
+  const endDate = new Date(row.endDate)/ 1000
+  const startDate = new Date(row.startDate)/ 1000
+  const currentDate = Date.now()/ 1000
+  const capReached = row.capUsed > row.cap
+
+  if (currentDate > startDate && currentDate < endDate){
+    status = capReached ? 'capReached' : 'active'
+  }
+  else if (currentDate < startDate)
+    status = 'pending'
+  else if (currentDate > endDate)
+    status = 'completed'
+
+  return {
+    id: row.id,
+    name: row.name,
+    startDate: startDate,
+    endDate: endDate,
+    distributionDate: row.distributionDate,
+    status: status,
+    actions: [], // TODO implement 
+    rewardEarned: { currency: 'OGN', amount: 0 } // TODO implement
```

Comment:
note that there is a campaign.currency field. so let's use that rather than hardcode OGN.

Reply:
Also rewardEarned will be by user so it's probably a method that won't be implemented in db.js but in a different layer ? In work-in-progress PR #1359 I have a Campaign class capable of evaluating rules per user. So maybe it should end up there ?

Reply:
Good point removing that in a separate PR